tianshou/utils/net/common.py

```diff
@@ -9,46 +9,134 @@
 def miniblock(
     inp: int,
     oup: int,
-    norm_layer: Optional[Callable[[int], nn.modules.Module]],
+    norm_layer: Optional[Callable[[int], nn.modules.Module]] = None,
+    activation: Optional[nn.modules.Module] = nn.ReLU,
 ) -> List[nn.modules.Module]:
     """Construct a miniblock with given input/output-size and norm layer."""
     ret: List[nn.modules.Module] = [nn.Linear(inp, oup)]
     if norm_layer is not None:
         ret += [norm_layer(oup)]
-    ret += [nn.ReLU(inplace=True)]
+    ret += [activation(inplace=True)]
     return ret
 
 
-class Net(nn.Module):
+class MLP(nn.Module):
     """Simple MLP backbone.
+    Create a MLP of size inp_shape*hidden_layer_size[0]*hidden_layer_size[1]...
+
+    :param hidden_layer_size: shape of MLP passed in as a list, not incluing
+        inp_shape.
+    :param norm_layer: use which normalization before activation, e.g.,
+        ``nn.LayerNorm`` and ``nn.BatchNorm1d``, defaults to no normalization.
+        You can also pass in a list of hidden_layer_number-1 normalization
+        modules to use different normalization module in different layers.
+        Default to empty list (no normalization).
+    :param activation: which activation to use after each layer, can be both
+        the same actvition for all layers if passed in nn.Module, or different
+        activation for different Modules if passed in a list.
+        Default to nn.ReLU.
+    """
+
+    def __init__(
+        self,
+        hidden_layer_size: Union[int, List[int]] = [],
```

Comment:
1. Never ever use mutable types as default value. 
2. Why not (128, 128) instead ? I still more usual as default

Reply:
because [] means, this Net do nothing, it outputs the input

Reply:
> because [] means, this Net do nothing, it outputs the input

Yes, and why would it be a good choice of default parameters ? I don't see why it would be better than no default.

Reply:
I add both `input_dim` and `output_dim` in MLP. Now if `hidden_sizes == []`, the MLP equals to a single `nn.Linear(input_dim, output_dim)`.

Reply:
Fine. Yet, using `[]` as default parameter is forbidden. See [this article](https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/) about it for instance.

So everywhere where `[]` is used as default should be replaced by either `None` or `()`, and typing accordingly (respectively `Optional[List[T]]` or `Sequence[T]`).